### PR TITLE
Added difficulty menu for single player games.

### LINKED ISSUE #69: Ability to play single player on Nightmare and Hell difficulty
Hi!

As an enhancement it would be nice if you could implement in the future the possibility to play single player on other difficulties than normal, ie nightmare and hell.
There was som weird ways to force the original into nightmare and hell difficulty so this should be possible.
This was how it was done:
https://www.youtube.com/watch?v=0fwXdDBviBQ

### PR DESCRIPTION
**Pull request by tylerbrawl**
With these changes, the user will be presented with the difficulty screen originally only seen in multiplayer games when either loading an old game or starting a new game with an _existing_ single player character. The menu does not appear when creating a single player character. (Besides, they would not be able to launch any of the other difficulties with a new character due to level requirements.)

Fixes #69

### CODE REVIEW

**Comment by tylerbrawl** [comment]
Also, I heard about an issue that causes the difficulty to reset in single-player games when generating a new dungeon. I have not experienced this issue myself with these changes, likely due to the modifications to `dvl::NetInit()` in commit 7906d1e.

File: SourceX/DiabloUI/selgame.cpp [hunk 1]
```diff
@@ -167,6 +169,69 @@ void selgame_GameSelection_Esc()
...
+
+	selgame_Free();
+
+	char *msg = NULL;
+	char *gameType = (isSP ? "single-player" : "multiplayer");
+	char *levelReq = (value == 1 ? "20" : "30");
+	char *diffType = (value == 1 ? "Nightmare" : "Hell");
+
+	char buffer[128];
+	sprintf_s(buffer, "Your character must reach level %s before you can enter a %s game of %s difficulty.", levelReq, gameType, diffType);
```
Comment (AJenbo) [inline SourceX/DiabloUI/selgame.cpp:191]: This looks to be windows only and probably why almost all builds fail on this PR

Comment (AJenbo) [reply SourceX/DiabloUI/selgame.cpp:191]: Don't worry to much about it though, I'm rebasing it and making some adjustments atm :)

**Comment by AJenbo** [comment]
Merged manually via https://github.com/diasurgical/devilutionX/commit/d7ca28ff8edc95a505fcae1addb6b98bb8887c32
Some adjustments to match the way the menu hack worked (always allow all diff for single player) and hellfire (save difficulty, show mode on pressinv v)

**Pull request closed without merging**